llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -370,163 +379,221 @@ static bool hasVertexIdx(const Constant &metaVal) {
+  assert(cast<ConstantInt>(getElemPtr->getOperand(1))->isZero() && "Non-zero GEP first index\n");
 
-    MDNode *metaNode = inOut->getMetadata(gSPIRVMD::InOut);
-    assert(metaNode);
-    auto inOutMetaVal = mdconst::dyn_extract<Constant>(metaNode->getOperand(0));
-
-    // If the input/output is arrayed, the outermost index might be used for vertex indexing
-    if (inOutTy->isArrayTy() && hasVertexIdx(*inOutMetaVal)) {
-      inOutTy = inOutTy->getArrayElementType();
-      vertexIdx = indexOperands[0];
-      ++operandIdx;
-      inOutMetaVal = cast<Constant>(inOutMetaVal->getOperand(1));
-    }
+  GlobalVariable *inOut = dyn_cast<GlobalVariable>(getElemPtr->getPointerOperand());
+  assert(!dyn_cast<GetElementPtrInst>(getElemPtr->getPointerOperand()) &&
+         "Chained GEPs should have been coalesced by SpirvLowerAccessChain.");
+  assert(inOut && "The root of the GEP should always be the global variable.");
 
-    loadValue = loadInOutMember(inOutTy, addrSpace, indexOperands, operandIdx, 0, inOutMetaVal, nullptr, vertexIdx,
-                                InterpLocUnknown, nullptr, &loadInst);
-  } else {
-    assert(isa<GlobalVariable>(loadSrc));
+  for (auto i = getElemPtr->idx_begin() + 1, e = getElemPtr->idx_end(); i != e; ++i)
+    indexOperands.push_back(toInt32Value(*i, &loadInst));
 
-    auto inOut = cast<GlobalVariable>(loadSrc);
-    auto inOutTy = inOut->getType()->getContainedType(0);
+  unsigned operandIdx = 0;
+  Value *vertexIdx = nullptr;
+  auto inOutTy = inOut->getType()->getContainedType(0);
 
-    MDNode *metaNode = inOut->getMetadata(gSPIRVMD::InOut);
-    assert(metaNode);
-    auto inOutMetaVal = mdconst::dyn_extract<Constant>(metaNode->getOperand(0));
+  MDNode *metaNode = inOut->getMetadata(gSPIRVMD::InOut);
+  assert(metaNode);
+  auto inOutMetaVal = mdconst::dyn_extract<Constant>(metaNode->getOperand(0));
+
+  // If the input/output is arrayed, the outermost index might be used for vertex indexing
+  if (inOutTy->isArrayTy() && hasVertexIdx(*inOutMetaVal)) {
+    inOutTy = inOutTy->getArrayElementType();
+    vertexIdx = indexOperands[0];
+    ++operandIdx;
+    inOutMetaVal = cast<Constant>(inOutMetaVal->getOperand(1));
+  }
 
-    loadValue = UndefValue::get(inOutTy);
+  Value *loadValue = loadInOutMember(inOutTy, addrSpace, indexOperands, operandIdx, 0, inOutMetaVal, nullptr, vertexIdx,
+                                     InterpLocUnknown, nullptr, &loadInst);
+  m_loadInsts.insert(&loadInst);
+  loadInst.replaceAllUsesWith(loadValue);
+}
 
-    if (inOutTy->isArrayTy() && hasVertexIdx(*inOutMetaVal)) {
-      auto elemTy = inOutTy->getArrayElementType();
-      auto elemMeta = cast<Constant>(inOutMetaVal->getOperand(1));
+// =====================================================================================================================
+// Handle "load" instructions.
+void SpirvLowerGlobal::handleLoadInst() {
+  auto handle = [&](const unsigned addrSpace) {
```

Comment:
`handle` sounds like this function does something. Maybe call it `shouldHandle`? Or `is…` (whatever this function checks)?